packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -245,101 +244,102 @@ export class FileSystemGraphObjectStore implements GraphObjectStore {
   async flushEntitiesToDisk(
     onEntitiesFlushed?: (entities: Entity[]) => Promise<void>,
   ) {
-    await this.lockOperation(() =>
-      pMap(
-        this.localGraphObjectStore.collectEntitiesByStep(),
-        async ([stepId, entities]) => {
-          const indexable = entities.filter((e) => {
-            const indexMetadata = this.getIndexMetadataForGraphObjectType({
-              stepId,
-              _type: e._type,
-              graphObjectCollectionType: 'entities',
-            });
-
-            if (typeof indexMetadata === 'undefined') {
-              return true;
-            }
-
-            return indexMetadata.enabled === true;
+    await this.lockOperation(async () => {
+      const entitiesByStep = this.localGraphObjectStore.collectEntitiesByStep();
+      let entitiesToUpload: Entity[] = [];
+      for (const [stepId, entities] of entitiesByStep) {
```

Comment:
Why did we move this from pMap to a plain function?

Reply:
Plain and simply there is no reason to divide the uploads we do. **There might be a point of doing it for writes to disk.**
But if a flushed batch of 6MB of entities gets into the function, do we really want to generate 100 uploads if the entities come from 100 different steps?

Reply:
That makes sense. In that case, we may want to alter the way this whole flushing process is done in the future. We could just remove the concept of steps in this. That can be future work though.